app/upgrades/historical.go

```diff
@@ -0,0 +1,84 @@
+package upgrades
+
+// This file is intended to keep old, historical upgrades in one place. It is advised to keep the future upgrades in the
+// separate file, and then move them to `historical.go` after a successful upgrade so the new nodes can still sync from
+// the genesis.
+
+// TODO_CONSIDERATION: after we verify `State Sync` is fully functional, we can hypothetically remove old upgrades from
+// the codebase, as the nodes won't have to execute upgrades and will download the "snapshot" instead. Some other
+// blockchain networks (such as `evmos`), remove the old upgrades from the codebase.
+
+import (
+	"context"
+
+	storetypes "cosmossdk.io/store/types"
+	upgradetypes "cosmossdk.io/x/upgrade/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	consensusparamtypes "github.com/cosmos/cosmos-sdk/x/consensus/types"
+	"github.com/pokt-network/poktroll/app/keepers"
+)
+
+// defaultMigrationsOnlyUpgradeHandler creates an update handler that only performs module's `ConsensusVersion`
+// change in blockchain state. Useful for performing upgrades that do no require additional state modifications, such as
+// parameter changes, data migrations, authz authorizations, etc. If **any** of these are needed, a new, version-specific,
+// upgrade handler should be created.
+func defaultMigrationsOnlyUpgradeHandler(
+	mm *module.Manager,
+	_ *keepers.Keepers,
+	configurator module.Configurator) upgradetypes.UpgradeHandler {
+	return func(ctx context.Context, plan upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+		return mm.RunMigrations(ctx, configurator, vm)
+	}
+}
+
+// An example of an upgrade that performs additional state changes.
+// Even when not changing `ConsensusVersion` of any modules, it still might be beneficial to create an upgrade
+// to signal to node runners utilizing `Cosmovisor` to automatically download and install the new binary.
+var Upgrade_Example = Upgrade{
+	PlanName:             "v0.0.0-Example",
+	CreateUpgradeHandler: defaultMigrationsOnlyUpgradeHandler,
+
+	// We can also add, rename and delete KVStores.
+	StoreUpgrades: storetypes.StoreUpgrades{},
```

Comment:
We can do this later, but it'd be great to have a concert example of this. Consider adding a TODO?

Reply:
Yep, and link to https://docs.cosmos.network/v0.50/learn/advanced/upgrade#add-storeupgrades-for-new-modules